File: Bachelor/Semester6-Finals/MPP - Projecting and Programing Methods/tReservation-Lab1CS/tReservation-Lab1CS/repository/IUserRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace tReservation_Lab1CS
{
    interface IUserRepository
    {
        //int size();
        //void save(User user);
        //void delete(int id);
        //void update(int id, User user);
        bool logIn(User user);
        //List<User> findAll();
    }
}

File: Bachelor/Semester6-Finals/MPP - Projecting and Programing Methods/tReservation-Lab1/src/main/java/repository/IUserRepository.java
package repository;

/**
 *
 */
public interface IUserRepository<Integer, User> {
    int size();
    void save(User entity);
    void delete(int id);
    void update(int id, User entity);
    User findOne(String userName);
    Iterable<User> findAll();
    boolean logIn(User user);
}